common/Timer.h

```diff
@@ -22,6 +23,15 @@ class Timer {
     ~Timer();
     FlowId getFlowEdge();
 
+    // TODO We could add more overloads for this if we need them (to create other kinds of Timers)
+    // We could also make this more generic to allow more sleep duration types.
+    static void timedSleep(const std::chrono::microseconds &sleep_duration, spdlog::logger &log, ConstExprStr name) {
+        Timer timer(log, name);
+        auto dur = std::chrono::duration<double, std::milli>(sleep_duration);
+        log.debug("{}: sleeping for {}ms", name.str, dur.count());
```

Comment:
isn't timer already logging?

Reply:
@dmitry-stripe The timer logs after the timer finishes. I wanted it to be clear that it's starting to sleep and finishing sleep.